File: github/github_accept_all_invites.rb
require 'net/http'
require 'uri'
require 'json'
require 'openssl'

token = ARGV[0]

uri = URI.parse('https://api.github.com/user/repository_invitations')

request = Net::HTTP::Get.new(uri)
request["Accept"] = "application/vnd.github.v3+json"
request["Authorization"] = "token #{token}"

req_options = {
  use_ssl: uri.scheme == "https",
}

response = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
  http.request(request)
end

result = JSON.parse(response.body) unless response.body.nil?

result.each do | invite |
  puts "Accepting invite for: #{invite['repository']['name']}"
  uri = URI.parse("https://api.github.com/user/repository_invitations/#{invite['id']}")

  request = Net::HTTP::Patch.new(uri)
  request["Accept"] = "application/vnd.github.v3+json"
  request["Authorization"] = "token #{token}"

  req_options = {
    use_ssl: uri.scheme == "https",
  }

  response = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
    http.request(request)
  end

  puts response
  puts ''
end

File: github/github_accept_all_invites.py
#!/usr/bin/env python3

from requests.auth import HTTPBasicAuth
import requests

pat = 'xxxxxxxxxx'
url = 'https://api.github.com/user/repository_invitations'

header = {'Authorization': 'token %s' % pat}

repository_invites = requests.get('https://api.github.com/user/repository_invitations', headers=header)

#print('response: ' + str(repository_invites.json()))

for repository_invite in repository_invites.json():
  repository_id = repository_invite.get('id')
  print('accepting invite for https://api.github.com/user/repository_invitations/%s' % str(repository_id))
  accept_invite = requests.patch('https://api.github.com/user/repository_invitations/'+ str(repository_id),
    data={}, headers=header )